python/cudf/cudf/pylibcudf_tests/test_string_find.py

```diff
@@ -0,0 +1,177 @@
+# Copyright (c) 2024, NVIDIA CORPORATION.
+
+import numpy as np
+import pyarrow as pa
+import pytest
+from utils import assert_column_eq
+
+import cudf._lib.pylibcudf as plc
+from cudf._lib.scalar import DeviceScalar
+
+
+@pytest.fixture()
+def plc_col():
+    data = pa.array(
+        ["AbC", "de", "FGHI", "j", "kLm", "nOPq", None, "RsT", None, "uVw"]
+    )
+    return plc.interop.from_arrow(data)
+
+
+@pytest.fixture
+def find_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+
+
+@pytest.fixture
+def contains_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["a", "d", "F", "j", "m", "q", None, "R", None, "w"])
+    )
+
+
+@pytest.fixture
+def starts_with_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["A", "d", "F", "j", "k", "n", None, "R", None, "u"])
+    )
+
+
+@pytest.fixture
+def ends_with_target_column():
+    return plc.interop.from_arrow(
+        pa.array(["C", "e", "I", "j", "m", "q", None, "T", None, "w"])
+    )
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_find(plc_col, target):
+    got = plc.strings.find.find(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
+    )
+
+    expected = pa.array(
+        [
+            elem.find(target) if elem is not None else None
+            for elem in plc.interop.to_arrow(plc_col).to_pylist()
+        ],
+        type=pa.int32(),
+    )
+
+    assert_column_eq(got, expected)
+
+
+def test_find_column(plc_col, find_target_column):
+    expected = pa.array(
+        [
+            elem.find(target) if elem is not None else None
+            for elem, target in zip(
+                plc.interop.to_arrow(plc_col).to_pylist(),
+                plc.interop.to_arrow(find_target_column).to_pylist(),
+            )
+        ],
+        type=pa.int32(),
+    )
+
+    got = plc.strings.find.find(plc_col, find_target_column, 0)
+    assert_column_eq(got, expected)
+
+
+@pytest.mark.parametrize("target", ["a", ""])
+def test_rfind(plc_col, target):
+    got = plc.strings.find.rfind(
+        plc_col, DeviceScalar(target, dtype=np.dtype("object")).c_value, 0, -1
```

Comment:
We shouldn't be create `cudf._lib.DeviceScalar` objects here. Can we just make a pylibcudf Scalar directly? For now, it's OK if that requires using the interop module. We'll eventually need to add proper constructors but for now removing usage of cudf from pylibcudf tests is a higher priority.

Also, while you're at it... the scalar could be another fixture :) You can parametrize the fixture directly [like this](https://docs.pytest.org/en/latest/how-to/fixtures.html#fixture-parametrize). This would again require ensuring that the same fixture can be used for all of the tests, but the way that you've written the tests I'm fairly certain that they should pass generically for arbitrary input data. This way you can also have a single place where you add new edge cases you want to test (like the empty string above).